pennylane/transforms/combine_global_phases.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Provides a transform to combine all ``qml.GlobalPhase`` gates in a circuit into a single one applied at the end.
+"""
+
+import pennylane as qml
+from pennylane.tape import QuantumScript, QuantumScriptBatch
+from pennylane.transforms import transform
+from pennylane.typing import PostprocessingFn
+
+
+@transform
+def combine_global_phases(tape: QuantumScript) -> tuple[QuantumScriptBatch, PostprocessingFn]:
+    """Combine all ``qml.GlobalPhase`` gates into a single ``qml.GlobalPhase`` operation.
+
+    This transform returns a new circuit including all the operations and measurements of the original input other than ``qml.GlobalPhase``.
+    A single ``qml.GlobalPhase`` operation is added at the end of the new output circuit.
+    The total global phase is computed as the algebraic sum of all the phases in the original circuit.
+
+    Args:
+        tape (QNode or QuantumScript or Callable): the input circuit to be transformed.
+
+    Returns:
+        qnode (QNode) or quantum function (Callable) or tuple[List[QuantumScript], function]:
+        the transformed circuit as described in :func:`qml.transform <pennylane.transform>`.
+
+    **Example**
+
+    Suppose we want to combine all the global phase gates in a given quantum circuit.
+    The ``combine_global_phases`` transform can be used to do this as follows:
+
+    .. code-block:: python3
+
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.transforms.combine_global_phases
+        @qml.qnode(dev)
+        def circuit():
+            qml.GlobalPhase(0.3, wires=0)
+            qml.PauliY(wires=0)
+            qml.Hadamard(wires=1)
+            qml.CNOT(wires=(1,2))
+            qml.GlobalPhase(0.46, wires=2)
+            return qml.expval(qml.X(0) @ qml.Z(1))
+
+    To check the result, let's print out the circuit:
+
+    >>> print(qml.draw(circuit)())
+    0: ──Y─────GlobalPhase(0.76)─┤ ╭<X@Z>
+    1: ──H─╭●──GlobalPhase(0.76)─┤ ╰<X@Z>
+    2: ────╰X──GlobalPhase(0.76)─┤
+    """
+
+    phi = 0
+    operations = []
+    for op in tape.operations:
+        if isinstance(op, qml.GlobalPhase):
+            phi += op.parameters[0]
+        else:
+            operations.append(op)
+
+    if qml.math.is_abstract(phi) or not qml.math.allclose(phi, 0):
```

Comment:
Whether we add a `GlobalPhase` should not even depend on `phi` anymore.

Reply:
@astralcai Is there a way to check whether the circuit contains at least a `GlobalPhase` without compromising `jit`-compatibility? Can I simply use a boolean variable?

Reply:
`jit` does not allow branching based on the value of an input parameter. In this case, a local boolean variable is totally fine.

Reply:
Resolved by commit https://github.com/PennyLaneAI/pennylane/pull/6686/commits/5cb0607a2c0654760073db366ecc51dfa0e660d9.